stacks/redpanda/deploy.sh

```diff
@@ -48,12 +48,5 @@ helm upgrade "$STACK" "$CHART" \
 MAX=50
 CURRENT=0
 
-until $(kubectl apply -f "https://raw.githubusercontent.com/vectorizedio/redpanda/$CHART_VERSION/src/go/k8s/config/samples/external_connectivity.yaml" >/dev/null 2>&1); do
-  CURRENT=$((CURRENT + 1))
-  sleep 1
-
-  if [ "$CURRENT" -gt "$MAX" ]; then
-    echo "FAILED: Webhook not ready, giving up."
-    exit 1
-  fi
-done
+kubectl wait pod -l app.kubernetes.io/instance=redpanda -n $NAMESPACE --timeout=10m --for condition=ready
```

Comment:
I wonder if there might be a race when pod is ready but the webhook is not yet listening 🤔

Reply:
The operator has readiness probe:

* https://github.com/vectorizedio/redpanda/blob/b18c4325f1131a63ceddebbe77aab97d08f1cf5c/src/go/k8s/config/manager/manager.yaml#L42-L47
* https://github.com/vectorizedio/redpanda/blob/b18c4325f1131a63ceddebbe77aab97d08f1cf5c/src/go/k8s/helm-chart/charts/redpanda-operator/templates/deployment.yaml#L77-L82

Reply:
but the port is bound before the webhook port is bound so there is a chance of race, but probably small one. https://github.com/kubernetes-sigs/controller-runtime/blob/c6f353f1f1d6880ca95915ff4e4d50971af2330f/pkg/manager/internal.go#L477